invenio_app_rdm/theme/assets/semantic-ui/js/invenio_app_rdm/administration/records/CompareRevisions.js

```diff
@@ -0,0 +1,185 @@
+    }
+  }
+
+  static contextType = NotificationContext;
+
+  computeDiff = (before, after) => {
+    const diff = this.differ.diff(before, after);
+    this.setState({ currentDiff: diff });
+  };
+
+  handleModalClose = () => {
+    const { actionCancelCallback } = this.props;
+    actionCancelCallback();
+  };
+
+  render() {
+    const { error, loading, currentDiff, allRevisions, srcRevision, targetRevision } =
+      this.state;
+
+    const options = Object.values(allRevisions).map((rev) => ({
+      key: rev.updated,
+      text: `${rev.updated} (${rev.revision_id})`,
+      value: rev,
+    }));
+
+    return (
+      <>
+        <Grid>
+          <Grid.Column width={6}>
+            <label htmlFor="source-revision">From</label>
+            <Dropdown
+              id="source-revision"
+              loading={loading}
+              placeholder="Source revision..."
+              fluid
+              selection
+              value={srcRevision}
+              onChange={(e, { value }) => {
+                this.setState({ srcRevision: value });
+              }}
+              options={options}
+            />
+          </Grid.Column>
+          <Grid.Column width={1} verticalAlign="middle">
+            <Icon name="exchange"></Icon>
+          </Grid.Column>
+          <Grid.Column width={6}>
+            <label htmlFor="target-revision">To</label>
+            <Dropdown
+              id="target-revision"
+              loading={loading}
+              placeholder="Target revision..."
+              fluid
+              selection
+              value={this.state.targetRevision}
+              onChange={(e, { value }) => {
+                this.setState({ targetRevision: value });
+              }}
+              options={options}
+            />
+          </Grid.Column>
+          <Grid.Column verticalAlign="bottom">
+            <Button
+              onClick={(e, { value }) => {
+                this.computeDiff(srcRevision, targetRevision);
+              }}
+            >
+              Compare
+            </Button>
+          </Grid.Column>
+        </Grid>
+        {error && (
+          <ErrorMessage
+            header={i18next.t("Unable to fetch revisions.")}
+            content={error}
+            icon="exclamation"
+            className="text-align-left"
+            negative
+          />
+        )}
+        <Modal.Content scrolling>
+          {loading && <p>Loading...</p>}
+          {!loading && currentDiff && (
+            <Segment>
+              <Viewer diff={currentDiff} {...this.viewerProps} />
+            </Segment>
+          )}
+        </Modal.Content>
+        <Modal.Actions>
+          <Button
+            onClick={this.handleModalClose}
+            floated="left"
+            disabled={loading}
+            loading={loading}
+            aria-label={i18next.t("Cancel revision comparison")}
+          >
+            Close
+          </Button>
+        </Modal.Actions>
+      </>
```

Comment:
this fragment is probably breaking the layout on actions. semantic expects direct descendant of modal